server/aws-lsp-codewhisperer/src/language-server/codeWhispererServer.ts

```diff
@@ -526,26 +515,31 @@ export const CodewhispererServerFactory =
             emitUserTriggerDecisionTelemetry(telemetry, session, timeSinceLastUserModification)
         }
 
-        const updateConfiguration = async () =>
-            lsp.workspace
-                .getConfiguration('aws.codeWhisperer')
-                .then(config => {
-                    if (config && config['includeSuggestionsWithCodeReferences'] === true) {
-                        includeSuggestionsWithCodeReferences = true
-                        logging.log('Configuration updated to include suggestions with code references')
-                    } else {
-                        includeSuggestionsWithCodeReferences = false
-                        logging.log('Configuration updated to exclude suggestions with code references')
-                    }
-                    if (config && config['shareCodeWhispererContentWithAWS'] === true) {
-                        codeWhispererService.shareCodeWhispererContentWithAWS = true
-                        logging.log('Configuration updated to share Amazon Q content with AWS')
-                    } else {
-                        codeWhispererService.shareCodeWhispererContentWithAWS = false
-                        logging.log('Configuration updated to not share Amazon Q content with AWS')
-                    }
-                })
-                .catch(reason => logging.log(`Error in GetConfiguration: ${reason}`))
+        const updateConfiguration = async () => {
+            try {
+                const qConfig = await lsp.workspace.getConfiguration(Q_CONFIGURATION_SECTION)
+                if (qConfig) {
+                    codeWhispererService.customizationArn = qConfig.customization ?? ''
```

Comment:
Is empty string is valid ARN value as fallback?

Reply:
I think instead of this, we can have a check like `if(qConfig && qConfig.customization)`